### DISCUSSION TITLE
Support for protocol buffer editions

### DISCUSSION THREAD

**Discussion by jqueuniet** [timostamm/protobuf-ts#663] [Q&A] [answered]
Hello,

The protocol buffer specs started evolving again recently, with a new versioning system called editions. Edition 2023 was released some time ago and a 2024 one is planned in the near future. Currently, using edition syntax with this project triggers an error.

```
main/v1/labels.proto: is an editions file, but code generator protoc-gen-ts hasn't been updated to support editions yet.  Please ask the owner of this code generator to add support or switch back to proto2/proto3.

See https://protobuf.dev/editions/overview/ for more information.--ts_out: Error: unsupported syntax "editions"
```

Is there a way to use editions with the current codebase that is not obvious, like by passing specific options or calling the plugin in a specific way? If not, is there any plan to add support to this project?

**Comment by timostamm** [comment, marked as answer]
Hey Johann, protobuf-es embeds minimal field info in the generated code - just the bare minimum to support serialization and custom options. In editions, features can be inherited to child elements, and the field info we're embedding is not well suited for this model. 

There is a Protobuf implementation that supports editions: https://github.com/bufbuild/protobuf-es. The plan for protobuf-ts is to switch code generation to the plugin framework of protobuf-es. This will shed a lot of dead weight from this repository, but there hasn't been much progress. Long term, it would be great to use more parts of protobuf-es under the hood, possibly including editions.

**Comment by jqueuniet** [comment]
Thanks for your answer. I assume the first paragraph is a typo and you're actually talking about protobuf-ts?
